lib/index.ts

```diff
@@ -139,8 +164,25 @@ export class LockAndCache {
    */
   private async _cacheGet<T>(key: CacheKey): Promise<T> {
     const keyStr = serializeKey(key);
-    const serializedValue = await this._cacheClient.get(keyStr);
-    if (serializedValue === null) {
+
+    // Check our caches. `_lruCache` returns `null` on a cache miss, and
+    // `_cacheClient` returns `undefined`. `== null` is the idiomatic way to
+    // check for either.
+    let serializedValue: string | undefined | null = this._lruCache.get(keyStr);
+
+    // It wasn't in our LRU, so let's try the network.
+    if (serializedValue == null) {
+      serializedValue = await this._cacheClient.get(keyStr);
+      if (serializedValue != null) {
```

Comment:
so can we cache null values?

Reply:
Yes. Notice that our caches only store `string` values, and `serializedValue` is declared as `string | undefined | null`. We deserialize the string values at the bottom of this function.

Reply:
We serialize `null` as `"null"` and `undefined` as `"undefined"`, and `[1]` as `"[1]"`. Etc.